data/lua/core.lua

```diff
@@ -232,6 +232,17 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 	wml.variable.set = wesnoth.set_variable
 	wml.variable.get_all = wesnoth.get_all_vars
 
+	local wesnoth_get_variable = wesnoth.get_variable
+	local wesnoth_set_variable = wesnoth.set_variable
```

Comment:
I notice that these local variables are the exact same as what was assigned above. Any particular reason for that? Why not just call wesnoth.get_variable directly, for example?

Reply:
Took me reading twice to see they're not duplicated right there.

get_wml_variable vs set_wml_variable has the difference g/s at the prime location so it's easiest to see.  if you want to be more different and more verbose, retrieve_* vs store_* is even less likely to be misread.

Reply:
The goal of having these lines is to replace a global variable access + subfield access with only one local variable access. Then again, if it feels like premature optimisation, I can remove those lines. I guess you decide.
Better keep local or better remove for code clarity?

Reply:
I don't think there's any point in assigning something to a local variable as an optimization. The reasoning I would use to decide whether or not to do it would be more like this — what if someone assigns their own function to `wesnoth.get_variable` (or `wml.variables.get`) which does some additional work on top of the basic variable fetch? Should your custom metatable do that extra work, or should it (as now) only provide the vanilla variable fetch?

If you have no preference, I'd lean towards not using a local variable.

Reply:
@CelticMinstrel on the other hand, no add-on should sanely do that because it might break other add-ons (effect is not hidden under unique namespace). But I agree what these local variables may confuse more than they help.